### PR TITLE
Add default_value parameter to get_env for use in make_launch_session

### LINKED ISSUE #1491: iOS/iPadOS/tvOS streaming broken on latest nightly
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your issue described in the documentation?

- [X] I have read the documentation

### Is your issue present in the nightly release?

- [X] This issue is present in the nightly release

### Describe the Bug

From Nightly https://github.com/LizardByte/Sunshine/actions/runs/5708257343 and onwards, it appears that streaming to Moonlight iOS/tvOS is non-functional.
20230801_053137000_iOS

### Expected Behavior

Expect stream to start as it does on the macOS client

### Additional Context

Last functioning Nightly in my testing appears to be https://github.com/LizardByte/Sunshine/actions/runs/5595679121

### Host Operating System

Windows

### Operating System Version

Windows 11 22H2 (22621.2070)

### Architecture

64 bit

### Sunshine commit or version

Starting from https://github.com/LizardByte/Sunshine/actions/runs/5708257343

### Package

Windows - installer

### GPU Type

Nvidia

### GPU Model

Geforce RTX 3080 10GB

### GPU Driver/Mesa Version

535.98

### Capture Method (Linux Only)

_No response_

### Config

```Shell
pkey = C:\Users\user\OneDrive\config\sunshine\credentials\key.pem
cert = C:\Users\user\OneDrive\config\sunshine\credentials\cert.pem
file_state = C:\Users\user\OneDrive\config\sunshine\state.json
file_apps = C:\Users\user\OneDrive\config\sunshine\apps.json
resolutions = [
    352x240,
    480x360,
    858x480,
    1280x720,
    1920x1080,
    1920x1200,
    3840x2160
]
channels = 2
```

### Apps

```json
{
    "env": {
        "PATH": "$(PATH);$(ProgramFiles(x86))\\Steam"
    },
    "apps": [
        {
            "name": "Desktop",
            "image-path": "desktop.png",
            "prep-cmd": [
                {
                    "do": "cmd \/c C:\\Users\\user\\OneDrive\\bin\\QRes.exe \/X:%SUNSHINE_CLIENT_WIDTH% \/Y:%SUNSHINE_CLIENT_HEIGHT%",
                    "undo": "C:\\Users\\user\\OneDrive\\bin\\qres.exe \/x:3440 \/y:1440 \/r:175"
                },
                {
                    "do": "C:\\Users\\user\\OneDrive\\bin\\nirsoft\\ControlMyMonitor.exe \/SetValue Primary 10 0",
                    "undo": "C:\\Users\\user\\OneDrive\\bin\\nirsoft\\ControlMyMonitor.exe \/SetValue Primary 10 0"
                }
            ],
            "exclude-global-prep-cmd": "false",
            "elevated": "false"
        },
        {
            "name": "Steam Big Picture",
            "detached": [
                "steam steam:\/\/open\/bigpicture"
            ],
            "image-path": "steam.png",
            "exclude-global-prep-cmd": "false",
            "elevated": "false"
        },
        {
            "name": "WUXGA (16:10)",
            "output": "",
            "cmd": "",
            "prep-cmd": [
                {
                    "do": "C:\\Users\\user\\OneDrive\\bin\\qres.exe \/x:1920 \/y:1200 \/r:120",
                    "undo": "C:\\Users\\user\\OneDrive\\bin\\qres.exe \/x:3440 \/y:1440 \/r:175"
                },
                {
                    "do": "C:\\Users\\user\\OneDrive\\bin\\nirsoft\\ControlMyMonitor.exe \/SetValue Primary 10 0",
                    "undo": "C:\\Users\\user\\OneDrive\\bin\\nirsoft\\ControlMyMonitor.exe \/SetValue Primary 10 0"
                }
            ],
            "image-path": "",
            "exclude-global-prep-cmd": "false",
            "elevated": "false"
        },
        {
            "name": "VGA (4:3)",
            "output": "",
            "cmd": "",
            "prep-cmd": [
                {
                    "do": "C:\\Users\\user\\OneDrive\\bin\\qres.exe \/x:1920 \/y:1440 \/r:175",
                    "undo": "C:\\Users\\user\\OneDrive\\bin\\qres.exe \/x:3440 \/y:1440 \/r:175"
                },
                {
                    "do": "C:\\Users\\user\\OneDrive\\bin\\nirsoft\\ControlMyMonitor.exe \/SetValue Primary 10 0",
                    "undo": "C:\\Users\\user\\OneDrive\\bin\\nirsoft\\ControlMyMonitor.exe \/SetValue Primary 10 0"
                }
            ],
            "image-path": "",
            "exclude-global-prep-cmd": "false"
        },
        {
            "name": "WQHD (16:9)",
            "output": "",
            "cmd": "",
            "prep-cmd": [
                {
                    "do": "C:\\Users\\user\\OneDrive\\bin\\qres.exe \/x:2560 \/y:1440 \/r:175",
                    "undo": "C:\\Users\\user\\OneDrive\\bin\\qres.exe \/x:3440 \/y:1440 \/r:175"
                },
                {
                    "do": "C:\\Users\\user\\OneDrive\\bin\\nirsoft\\ControlMyMonitor.exe \/SetValue Primary 10 0",
                    "undo": "C:\\Users\\user\\OneDrive\\bin\\nirsoft\\ControlMyMonitor.exe \/SetValue Primary 10 0"
                }
            ],
            "image-path": "",
            "exclude-global-prep-cmd": "false"
        },
        {
            "name": "Steam Deck",
            "output": "",
            "cmd": "",
            "exclude-global-prep-cmd": "false",
            "elevated": "false",
            "prep-cmd": [
                {
                    "do": "C:\\Users\\user\\OneDrive\\bin\\qres.exe \/x:1280 \/y:800 \/r:120",
                    "undo": "C:\\Users\\user\\OneDrive\\bin\\qres.exe \/x:3440 \/y:1440 \/r:175",
                    "elevated": "false"
                },
                {
                    "do": "C:\\Users\\user\\OneDrive\\bin\\nirsoft\\ControlMyMonitor.exe \/SetValue Primary 10 0",
                    "undo": "C:\\Users\\user\\OneDrive\\bin\\nirsoft\\ControlMyMonitor.exe \/SetValue Primary 10 0",
                    "elevated": "false"
                }
            ],
            "image-path": ""
        }
    ]
}
```

### Relevant log output

```shell
N/A. Not sure how to gather this from iOS/tvOS. Happy to do so, if there's some guidance on this front.
```

### PR DESCRIPTION
**Pull request by Xythus**
## Description
Add an additional parameter to get_env to optionally provide a default value.
This value will be used instead of throwing an out of range exception when the requested argument is not found in args.
The addition of this argument now allows sessions to launch when a client does not provide all the requested arguments. An example of this is the Moonlight iOS client, which does not seem to provide a UUID.
I have only used this new addition in make_launch_session as this is where I encountered the issue. Perhaps other places where get_env is used could make use of this. :)

### Issues Fixed or Closed
Fixes an issue where sessions would not be able to launch if the client does not provide all the requested arguments.

- Fixes #1491

## Type of Change
- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency update (updates to dependencies)
- [ ] Documentation update (changes to documentation)
- [ ] Repository update (changes to repository files, e.g. `.github/...`)

## Checklist
- [ ] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated the in code docstring/documentation-blocks for new or existing methods/components

## Branch Updates
LizardByte requires that branches be up-to-date before merging. This means that after any PR is merged, this branch
must be updated before it can be merged. You must also
[Allow edits from maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [X] I want maintainers to keep my branch updated

### CODE REVIEW

File: src/nvhttp.cpp [hunk 1]
```diff
@@ -278,17 +282,17 @@ namespace nvhttp {
...
     }
-    launch_session.unique_id = (get_arg(args, "uniqueid"));
-    launch_session.uuid = (get_arg(args, "uuid"));
-    launch_session.appid = util::from_view(get_arg(args, "appid"));
-    launch_session.enable_sops = util::from_view(get_arg(args, "sops"));
-    launch_session.surround_info = util::from_view(get_arg(args, "surroundAudioInfo"));
-    launch_session.gcmap = util::from_view(get_arg(args, "gcmap"));
+    launch_session.unique_id = (get_arg(args, "uniqueid", "0123456789ABCDEF"));
+    launch_session.uuid = (get_arg(args, "uuid", "00000000000000000000000000000000"));
+    launch_session.appid = util::from_view(get_arg(args, "appid", "0123456789"));
```
Comment (ReenigneArcher) [inline src/nvhttp.cpp:287]: I don't think we should provide anything except an empty string as the default.

```suggestion
    launch_session.unique_id = (get_arg(args, "uniqueid", ""));
    launch_session.uuid = (get_arg(args, "uuid", ""));
    launch_session.appid = util::from_view(get_arg(args, "appid", ""));
```

Comment (Xythus) [reply src/nvhttp.cpp:287]: Sure, that works. I'm not familiar with the internal workings of Sunshine and wanted to make sure the provided defaults at least somewhat represent the actual values as to not trip up any logic elsewhere. You're probably more familiar with the risks.

Comment (ReenigneArcher) [reply src/nvhttp.cpp:287]: or `"None"`, `"unknown"`... something like that

Comment (Xythus) [reply src/nvhttp.cpp]: I changed them to `"unknown"`. Does that work?

File: src/nvhttp.cpp [hunk 2]
```diff
@@ -278,17 +282,17 @@ namespace nvhttp {
...
-    launch_session.uuid = (get_arg(args, "uuid"));
-    launch_session.appid = util::from_view(get_arg(args, "appid"));
-    launch_session.enable_sops = util::from_view(get_arg(args, "sops"));
-    launch_session.surround_info = util::from_view(get_arg(args, "surroundAudioInfo"));
-    launch_session.gcmap = util::from_view(get_arg(args, "gcmap"));
+    launch_session.unique_id = (get_arg(args, "uniqueid", "unknown"));
+    launch_session.uuid = (get_arg(args, "uuid", "unknown"));
+    launch_session.appid = util::from_view(get_arg(args, "appid", "unknown"));
+    launch_session.enable_sops = util::from_view(get_arg(args, "sops", "0"));
+    launch_session.surround_info = util::from_view(get_arg(args, "surroundAudioInfo", "0"));
```
Comment (cgutman) [inline src/nvhttp.cpp:289]: `surroundAudioInfo`  default value should be `196610` which corresponds to stereo channel count and channel mask.

File: src/nvhttp.cpp [hunk 3]
```diff
@@ -267,8 +271,8 @@ namespace nvhttp {
     rtsp_stream::launch_session_t launch_session;
 
     launch_session.host_audio = host_audio;
-    launch_session.gcm_key = util::from_hex<crypto::aes_t>(get_arg(args, "rikey"), true);
-    std::stringstream mode = std::stringstream(get_arg(args, "mode"));
+    launch_session.gcm_key = util::from_hex<crypto::aes_t>(get_arg(args, "rikey", "0"), true);
```
Comment (cgutman) [inline src/nvhttp.cpp:274]: `rikey` and `rikeyid` are both required arguments, so you can drop the default value for those.

File: src/nvhttp.cpp [hunk 4]
```diff
@@ -278,17 +282,17 @@ namespace nvhttp {
...
+    launch_session.unique_id = (get_arg(args, "uniqueid", "unknown"));
+    launch_session.uuid = (get_arg(args, "uuid", "unknown"));
+    launch_session.appid = util::from_view(get_arg(args, "appid", "unknown"));
+    launch_session.enable_sops = util::from_view(get_arg(args, "sops", "0"));
+    launch_session.surround_info = util::from_view(get_arg(args, "surroundAudioInfo", "0"));
+    launch_session.gcmap = util::from_view(get_arg(args, "gcmap", "0"));
     launch_session.enable_hdr = 0;
     if (args.find("enableHdr"s) != std::end(args)) {
-      launch_session.enable_hdr = util::from_view(get_arg(args, "enableHdr"));
+      launch_session.enable_hdr = util::from_view(get_arg(args, "enableHdr", "0"));
```
[the remark this answers is not in the document]
Comment (Xythus) [reply src/nvhttp.cpp]: Done!

File: src/nvhttp.cpp (hunk 3 above)
Comment (Xythus) [reply src/nvhttp.cpp]: Done!

**Review by cgutman — APPROVED** [review]

**Pull request merged**